src/ifcconvert/XmlSerializer.cpp

```diff
@@ -328,10 +368,20 @@ void XmlSerializer::finalize() {
 		}
 	}
 
+    // Layer assignments
+    IfcPresentationLayerAssignment::list::ptr layer_assignments = file->entitiesByType<IfcPresentationLayerAssignment>();
+    for (IfcPresentationLayerAssignment::list::it it = layer_assignments->begin(); it != layer_assignments->end(); ++it) {
+        // IfcPresentationLayerAssignments don't have GUIDs (only optional Identifier) so use name as the ID
+        ptree node;
+        node.put("<xmlattr>.id", "#" + (*it)->Name());
```

Comment:
The `"#"` is not part of the `id` (so should be removed here), but as part of the `href` it signifies that the link points to a node in the same document. The xlink w3c spec is a bit too verbose to me now, but the example here [1] illustrates this.

[1] https://en.wikipedia.org/wiki/XLink#Simple_links